### PR TITLE
Blog post whats new in Camel 3.7

### PR DESCRIPTION
**Pull request by davsclaus**
Feedback is welcome. You are also welcome to add new section if I missed anything noteworthy to cover. And for typos and grammar, then go ahead and fix them asap.

### CODE REVIEW

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+The next planned LTS release is 3.10 scheduled towards summer 2021.
+
+
+## So what's in this release?
+
+This release introduces a set of new features and noticeable improvements that we will cover in this blog post.
+
+### Pre compiled languages
+
+We are continued our avenue of making Camel faster and smaller.
```
Comment (gzurowski) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:21]: "We continued our avenue..." (remove "are")

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+This brings up to lightweight mode.
+
+
+### Lightweight mode
+
+We started an experiment earlier with a lightweight mode. With the separation of the model from the processors,
+then we have a great step forward, which allowed us to make the lightweight mode available for end users to turn on.
+
+In lightweight mode, Camel will after startup, then remove all references to the model, and other bits, that all together
```
Comment (gzurowski) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:118]: Not clear, could you rephrase?

**Review by gzurowski — APPROVED** [review]
LGTM 👍 , some minor comments added.

**Comment by davsclaus** [comment]
Thanks Gregor for the feedback

**Review by oscerd — APPROVED** [review]

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+### Compiled Simple
+
+The [csimple language](https://camel.apache.org/components/latest/languages/csimple-language.html)
+is parsed into regular Java source code and compiled together with all the other source code,
+or compiled once during bootstrap via jOOR.
+
+{{< image "csimple-compiled.png" "Compiled Simple Language" >}}
+
+In a nutshell where compiled simple excels over simple language is primary around dynamic OGNL method calls.
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:38]: I had to read several times to understand this sentence.

What about something like:
In a nutshell, `Compiled Simple` excels primarily over simple language when using dynamic Object-Graph Navigation Language ([OGNL](https://en.wikipedia.org/wiki/OGNL) ) method calls..

and wtih a link to

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+             ${exchangeProperty.user.getAge()} > 11</csimple>
+
+yields a dramatic 100 times performance improvement in reduced cpu usage as shown in the screenshot:
+
+{{< image "simple-csimple-performance.png" "Simple vs Compiled Simple" >}}
+
+For more information about the compiled simple language and further break down of performance improvements
+then read [Claus blog post](http://www.davsclaus.com/2020/12/apache-camel-37-compiled-simple.html).
+
+We have provided two small examples that demonstrate csimple as pre compiled and as runtime compiled during bootstrap.
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:57]:
```suggestion
We have provided two small examples that demonstrate `csimple` as pre compiled and as runtime compiled during bootstrap.
```

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+In this release we untangled model, reifier and processors.
+
+This is a great achievement which allows us to take this even further with design time vs runtime.
+
+    Model    ->    Reifier   ->   Processor
+    (startup)      (startup)      (runtime)
+
+The model is the structure of the DSL which you can think of as _design time_ specifying your Camel routes.
+The model is executed once during startup and via the reifier (factory) the runtime EIP processors is cretated.
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:104]:
```suggestion
The model is executed once during startup and via the reifier (factory) the runtime EIP processors is created.
```

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+### Lightweight mode
+
+We started an experiment earlier with a lightweight mode. With the separation of the model from the processors,
+then we have a great step forward, which allowed us to make the lightweight mode available for end users to turn on.
+
+In lightweight mode Camel removes all references to the model after startup 
+which causes the JVM to be able to garbage collect all model objects and unload classes, freeing up memory.
+
+After this it's no longer possible to dynamic add new Camel routes. The lightweight mode is intended for
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:121]:
```suggestion
After this it's no longer possible to dynamically add new Camel routes. The lightweight mode is intended for
```

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+microservice/serverless architectures, with a _closed world_ assumption.
+
+
+### Autowiring components
+
+The Camel components is now capable of autowiring by type. For example the AWS SQS components can automatic
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:127]:
```suggestion
The Camel components is now capable of autowiring by type. For example the [AWS SQS](https://camel.apache.org/components/latest/aws-sqs-component.html) components can automatically
```

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+lookup in the registry if there is single instance of `SqsClient`, and then pre configure itself.
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:128]:
```suggestion
lookup in the registry if there is a single instance of `SqsClient`, and then pre configure itself.
```

**Review by aldettinger — APPROVED** [review]

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+
+We have marked up in the Camel documentation which component options supports this by showing *Autowired* in bold
+in the description.
+
+
+### Salesforce fixes
+
+Our recent Camel committer Jeremy Ross did great work to improve and fix bugs in the camel-salesforce component.
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:136]:
```suggestion
Our recent Camel committer [Jeremy Ross](https://github.com/jeremyross) did great work to improve and fix bugs in the [camel-salesforce component](https://camel.apache.org/components/latest/salesforce-component.html).
```

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+### VertX Kafka Component
+
+A new Kafka component has been developed that uses the Vert.X Kafka Java Client which allows us to use all of its
+features, and also its robustness and stability. 
+
+The [camel-vertx-kafka](https://camel.apache.org/components/latest/vertx-kafka-component.html) component is intended
+to be (more) feature complete with the existing camel-kafka component. We will continue this work for the next couple of Camel releases.
+
+The existing camel-kafka component has a few issues reported, which we want to fix in the upcoming releases,
+and at the same time we can offer this new camel-kafka-component as an alternative and potential a better component in the future.
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:149]: you mean "we can offer this new camel-vertx-kafka component"?

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+
+
+### DataSonnet
+
+The new [camel-datasonnet](https://camel.apache.org/components/latest/languages/datasonnet-language.html) component,
+is to be used for data transformation using the DataSonnet standard.
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:155]:
```suggestion
is to be used for data transformation using the [DataSonnet](https://datasonnet.com/) standard.
```

File: content/blog/2020/12/Camel37-Whatsnew/index.md [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+### Spring Boot
+
+We have upgraded to Spring Boot 2.4.
+
+
+### New components
+
+This release has a number of new components, data formats and languages:
+
+- AtlasMap: Transforms the message using an AtlasMap transformation
```
Comment (apupier) [inline content/blog/2020/12/Camel37-Whatsnew/index.md:170]:
```suggestion
- [AtlasMap](https://camel.apache.org/components/latest/atlasmap-component.html): Transforms the message using an [AtlasMap](https://www.atlasmap.io/) transformation
```

**Comment by davsclaus** [comment]
Thanks @apupier for the feedback
